growattServer/__init__.py

```diff
@@ -116,12 +116,15 @@ def login(self, username, password, is_password_hashed=False):
             'userName': username,
             'password': password
         })
-        data = json.loads(response.content.decode('utf-8'))['back']
-        if data['success']:
-            data.update({
-                'userId': data['user']['id'],
-                'userLevel': data['user']['rightlevel']
-            })
+        try:
+            data = json.loads(response.content.decode('utf-8'))['back']
+            if data['success']:
+                data.update({
+                    'userId': data['user']['id'],
+                    'userLevel': data['user']['rightlevel']
+                })
+        except:
+            data = {'success': False}
```

Comment:
Since 1.3.1 the http client manages correct errors if the response has an unexpected http code
```suggestion
        data = json.loads(response.content.decode('utf-8'))['back']
        if data['success']:
            data.update({
                'userId': data['user']['id'],
                'userLevel': data['user']['rightlevel']
            })
```